controllers/componentversion_controller.go

```diff
@@ -416,6 +411,12 @@ func (r *ComponentVersionReconciler) checkVersion(ctx context.Context, octx ocm.
 		latest,
 	)
 
+	if latestSemver.Equal(current) {
```

Comment:
Hey @Skarlso 

I am worried that this might over simplify things.   We have seen that the folks that manage the artifactory we use sometimes 'lose' images ( don't know how, but it happens ).

This is why the logic I shared with you included a test of whether the current version was matching the ComponentVersion's semver check.   I was wanting to allow for rollback if it was being forced by the semver check but not just because an image temporarily went missing from artifactory.

Reply:
That happens in a different section. Here: https://github.com/open-component-model/ocm-controller/blob/main/pkg/ocm/ocm.go#L423

Is this what you mean? So at that point the latestVALID semver is being considered.

Reply:
Hey 
Yes and no :)

I am asking for that validity check to be performed against the currently deployed version so that we rollback only if the currently deployed version is not considered valid anymore.   So the check you pointed to is the thing I am thinking of however it isn't being applied against the current deployed version.

So here is scenario where a roll back should _not_ happen 
-  semVer is ">=1.0.0"
-   2.0.0 is currently deployed
- on reconcile repository has 1.0.0 and 1.5.0 but somehow lost 2.0.0 ( so latest valid from repo is 1.5.0 )

However here is where it should roll back 
-  semVer is ">=1.0.0,!=2.0.0"
-   2.0.0 is currently deployed
- on reconcile repository has 1.0.0 and 1.5.0.  If it has 2.0.0 doesn't matter because semVer filters it out

And of course this would simply roll forward 
-  semVer is ">=1.0.0,!=2.0.0"
-   2.0.0 is currently deployed
- on reconcile repository has 1.0.0 and 1.5.0 and 2.5.0.  If it has 2.0.0 doesn't matter because semVer filters it out

So in the first case we aren't rolling back just because 2.0.0 is no longer available from the repository.    On the other hand in the second case we are back because the latest from the repository 1.5.0 _and_ the currently deployed version is no longer passing the semVer check

The reason we need this is cause somehow our artifactory is sometimes losing images temporarily.    In this case we don't want our deployments to rollback just because of some glitch.     On the other hand, if we determined a version, 2.0.0 in the above example, is bad and we need to quickly go back to the last good version then we want to be able to simply exclude it via the semVer check.

Reply:
Okay yeah. I think I understand. Thanks.